data-prepper-plugins/cloudwatch-metrics-source/src/test/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsWorkerTest.java

```diff
@@ -0,0 +1,157 @@
+import org.opensearch.dataprepper.plugins.buffer.blockingbuffer.BlockingBuffer;
+import org.opensearch.dataprepper.plugins.source.configuration.DimensionConfig;
+import org.opensearch.dataprepper.plugins.source.configuration.DimensionsListConfig;
+import org.opensearch.dataprepper.plugins.source.configuration.MetricDataQueriesConfig;
+import org.opensearch.dataprepper.plugins.source.configuration.MetricsConfig;
+import org.opensearch.dataprepper.plugins.source.configuration.NamespaceConfig;
+import org.opensearch.dataprepper.plugins.source.configuration.NamespacesListConfig;
+import software.amazon.awssdk.services.cloudwatch.CloudWatchClient;
+import software.amazon.awssdk.services.cloudwatch.model.Dimension;
+import software.amazon.awssdk.services.cloudwatch.model.GetMetricDataRequest;
+import software.amazon.awssdk.services.cloudwatch.model.GetMetricDataResponse;
+import software.amazon.awssdk.services.cloudwatch.model.MetricDataResult;
+import software.amazon.awssdk.services.cloudwatch.model.MetricStat;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class CloudwatchMetricsWorkerTest {
+
+    private CloudWatchClient cloudWatchClient;
+
+    private BufferAccumulator<Record<Event>> bufferAccumulator;
+
+    private CloudwatchMetricsSourceConfig cloudwatchMetricsSourceConfig;
+
+    private Collection<Dimension> dimensionCollection;
+
+    private NamespacesListConfig namespacesListConfig;
+
+    private NamespaceConfig namespaceConfig;
+
+    private PluginMetrics pluginMetrics;
+
+    private GetMetricDataResponse getMetricDataResponse;
+
+    private MetricDataQueriesConfig metricDataQueriesConfig;
+
+    private MetricsConfig metricsConfig;
+
+    private DimensionsListConfig dimensionsListConfig;
+
+    private SourceCoordinator sourceCoordinator;
+
+    MetricDataResult metricDataResultMock = MetricDataResult.builder()
+            .id("BucketSizeBytes")
+            .statusCode("SUCCESS")
+            .values(1.10)
+            .build();
+
+    @BeforeEach
+    public void setup(){
+        final String namespaceName = "AWS/S3";
+        this.sourceCoordinator = mock(SourceCoordinator.class);
+        this.cloudWatchClient = mock(CloudWatchClient.class);
+        this.dimensionCollection = new ArrayList<>();
+        this.dimensionCollection.add(Dimension.builder().name("StorageType").value("StandardStorage").build());
```

Comment:
I think you are trying to force `dimensionCollection` to be a field to help with testing. This may be an indication that you should split some of the logic into another class. That other class could be easier to test.

Reply:
we have changed some implementation of CloudWatchMetricWorker constructor so  no need of  separate logic.